feature_engine/dataframe_checks.py

```diff
@@ -2,25 +2,64 @@
 # License: BSD 3 clause
 
 import pandas as pd
+from typing import Union, List
+
+def _is_dataframe(X: pd.DataFrame) -> pd.DataFrame:
+    """
+    Checks if the input is a DataFrame.
+    Creates the copy of the initial DataFrame.
+
+    Args:
+        X: Argument to perform check against
+
+    Raises:
+        TypeError: If the input is not the Pandas DataFrame
+
+    Returns:
+        The copy of initial DataFrame.
+        Important not to transform the original dataset.
+    """
 
-def _is_dataframe(X):
-    # checks if the input is a dataframe. Also creates a copy,
-    # important not to transform the original dataset.
     if not isinstance(X, pd.DataFrame):
         raise TypeError("The data set should be a pandas dataframe")
+
     return X.copy()
 
 
-def _check_input_matches_training_df(X, reference):
-    # check that dataframe to transform has the same number of columns
-    # that the dataframe used during fit method
+def _check_input_matches_training_df(X: pd.DataFrame, reference: int) -> None:
+    """
+    Check that DataFrame to transform has the same number
+    of columns that the DataFrame used during fit method.
+
+    Args:
+        X: Pandas DataFrame to perform comparison
+        reference: Number of columns
+
+    Raises:
+        ValueError: If number of columns is not the same
+
+    Returns:
+        None
+    """
+
     if X.shape[1] != reference:
         raise ValueError('The number of columns in this data set is different from that of the train set used during'
                          'the fit method')
+
     return None
 
 
-def _check_contains_na(X, variables):
+def _check_contains_na(X: pd.DataFrame, variables: Union[str, List[str]]):
+    """
```

Comment:
variables here should always be a list.

Reply:
@solegalli this function works when the input is only string due to Padas way to select columns. Is it necessary to restrict ```variables``` to be the only list?

Reply:
Tests don't fail and let the user be able to check only one variable against the null values. If after all the changes I have planned to do the test will fail, then I will notify you and propose the fix for this function.